### DISCUSSION TITLE
Middleware after Endpoint

### DISCUSSION THREAD

**Discussion by KBroichhausen** [TypedProject/ts-express-decorators#1413] [Q&A] [answered]
I want to add a Middleware after Endpoints to log specific results. I definetly don't want to modify the existing code except for the decorators.

Now, what I understand from previous Issues that `UseAfter` is only executed when there was no response returned. Is there any way to execute Middleware even if the response was returned?
Not executing the `UseAfter` makes sense because you can not change the response anymore but I just need reading acces, so in my use case it would make sense to log application specific results.

The documentation https://tsed.io/docs/middlewares.html#call-sequences is quite irritating for me.
Why there are two executions of `UseAfter` (one as green, one as purple)?
What is `UseAfterEach`? It seems like it only occurs in that one document and not in source code (or simply haven't found it).

**Comment by Romakita** [comment, marked as answer]
@KBroichhausen you can use Interceptor instead of a middleware ;)

**Reply by Romakita** [reply]
There are twice useAfter because one is registered on controller level and the second is registered on endpoint level

**Reply by Romakita** [reply]
I’ll simplify the documentation and the schema to be more  understable

**Reply by KBroichhausen** [reply]
Thanks. Since I need the body of the response, it get's a little bit hacky because Express has no accessable field for that. I write it to the `locals` field of the Response with another Middleware for all `send` calls.
The logging `Interceptor` who needs this is only attached to two endpoints.
